pyomo/contrib/doe/fim_doe.py

```diff
@@ -1633,21 +1758,22 @@ def __init__(self, para_name, measurement_variables, measurement_timeset,  prior
         self.para_name = para_name
         self.measurement_variables = measurement_variables
         self.measurement_timeset = measurement_timeset
-        #self.measurement_extra_index = measurement_extra_index
+        self.flatten_all_measure = flatten_all_measure
         self.prior_FIM = prior_FIM
         self.store_FIM = store_FIM
         self.scale_constant_value = scale_constant_value
         self.fim_scale_constant_value = scale_constant_value ** 2
         self.max_condition_number = max_condition_number
         self.verbose = verbose
 
-    def calculate_FIM(self, jaco_information, dv_values, jaco_involved_name=None, jaco_involved_extra_index=None, result=None):
+    def calculate_FIM(self, jaco_information, jaco_3D, dv_values, jaco_involved_name=None, jaco_involved_extra_index=None, result=None):
```

Comment:
Is jaco_info and jaco_3D redundant? Do you need both?

Reply:
We should standardize on one general representation for the Jacobian that works for all use cases. This will make the code much easier to maintain.